File: Number Factorization/factorization.cs
using System;
using System.Collections.Generic;

namespace Factorization
{
    public class Factorization
    {
        static void Main(string[] args)
        {
            List<int> factors = new List<int>();
            Console.WriteLine("Enter a number: ");
            int n = int.Parse(Console.ReadLine());

            while (n != 1)
            {
                for (int i = 2; i <= n; i++)
                {
                    if (n % i == 0)
                    {
                        n /= i;
                        factors.Add(i);
                        break;
                    }
                }
            }

            Console.Write("n = ");
            for (int j = 0; j < factors.Count; j++)
            {
                Console.Write(factors[j]);
                if (j < factors.Count - 1)
                {
                    Console.Write(" * ");
                }
            }
            Console.WriteLine();
        }
    }
}

File: Number Factorization/factorization.cpp
#include <iostream>
#include <vector>

void primeFactors(int n, std::vector<int> &factors) {
    while (n % 2 == 0) {
        factors.push_back(2);
        n = n / 2;
    }

    for (int i = 3; i * i <= n; i = i + 2) {
        while (n % i == 0) {
            factors.push_back(i);
            n = n / i;
        }
    }

    if (n > 2) {
        factors.push_back(n);
    }
}

int main() {
    int n;
    std::cout << "Type a number and press enter: " << std::endl;
    std::cin >> n;

    std::vector<int> factors;

    primeFactors(n, factors);

    std::cout << n << " = ";
    for (size_t i = 0; i < factors.size(); i++) {
        if (i != 0) {
            std::cout << " * ";
        }
        std::cout << factors[i];
    }
    std::cout << std::endl;

    return 0;
}